docs/16.MIGRATION-GUIDE.md

```diff
@@ -0,0 +1,772 @@
+# Migration guide
```

Comment:
Since it's a new document this [script](https://github.com/jerryscript-project/jerryscript/blob/master/tools/update-webpage.sh#L40) should be updated as well.